src/inmanta/agent/agent.py

```diff
@@ -142,9 +142,14 @@ async def _execute(self, ctx: handler.HandlerContext, requires: Dict[ResourceVer
                     self.resource,
                     requires,
                 )
+                if ctx.status is None:
+                    ctx.set_status(const.ResourceState.deployed)
```

Comment:
I'm not really familiar with this part of the code, so this looks a bit unintuitive, but based on the tests it looks fine